web3/providers/base.py

```diff
@@ -11,24 +11,23 @@
 
 
 class BaseProvider(object):
-    _middlewares = None
+    _middlewares = []
```

Comment:
This will/can leak across instances of the `Provider`.  Since lists are mutable, every instance will by default share this instance of this list.  Current implementation won't actual fail, but it's still bad mojo.

Reply:
Yeah, it's super defensive, since no one should be messing with it directly, but I'm on board. Easy enough to change to `()`